e2e-pw/src/oss/specs/sidebar/sidebar.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { test as base, expect } from "src/oss/fixtures";
+import { GridPom } from "src/oss/poms/grid";
+import { SidebarPom } from "src/oss/poms/sidebar";
+import { getUniqueDatasetNameWithPrefix } from "src/oss/utils";
+
+const datasetName = getUniqueDatasetNameWithPrefix("smoke-quickstart");
+
+const test = base.extend<{ sidebar: SidebarPom; grid: GridPom }>({
+  sidebar: async ({ page }, use) => {
+    await use(new SidebarPom(page));
+  },
+  grid: async ({ page }, use) => {
+    await use(new GridPom(page));
+  },
+});
+
+test.describe("sidebar-filter-visibility", () => {
+  test.beforeAll(async ({ fiftyoneLoader }) => {
+    await fiftyoneLoader.loadZooDataset("quickstart", datasetName, {
+      max_samples: 5,
+    });
+  });
+
+  test.beforeEach(async ({ page, fiftyoneLoader }) => {
+    await fiftyoneLoader.waitUntilLoad(page, datasetName);
+    // always fold tags and metaData groups
+    await page.click('[title="TAGS"]');
+    await page.click('[title="METADATA"]');
+  });
+
+  test("In grid, select a label filter works", async ({ grid, sidebar }) => {
+    // do not show prediction labels
+    await sidebar.clickFieldCheckbox("predictions");
+
+    // select bottlel in ground_truth.detections.label
+    await sidebar.clickFieldDropdown("ground_truth");
+    await sidebar.getLabelFromList(
+      "ground_truth.detections.label",
+      ["bottle"],
+      "select-detections-with-label"
+    );
+
+    // verify the number of samples in the result
+    await grid.assert.waitForEntryCountTextToEqual("1 of 5 samples");
+
+    await grid.assert.delay(1000);
```

Comment:
1. Why not `page.waitForTimeout` in `delay()` impl?
2. What are we waiting on? Is there an alternative way?